dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/runner/SubProcessTaskExecThread.java

```diff
@@ -112,6 +112,31 @@ private void updateParentProcessState(){
         this.processInstance.setState(parentProcessInstance.getState());
     }
 
+    /**
+     * wait for the establishment of sub-process-instance
+     * @throws InterruptedException
+     */
+    private void waitSubProcessEstablishment() throws InterruptedException {
+        if (subProcessInstance == null) {
+            while (Stopper.isRunning()) {
```

Comment:
i think it's better to merge this 'while' into the other.

Reply:
Restored.

Reply:
the restore caused another ridiculous history code smell...🙄